cirq/experiments/t2_decay_experiment.py

```diff
@@ -214,13 +307,21 @@ def _expectation(self, data) -> pd.DataFrame:
         measurement is all ones and -1 if the measurement is all zeros.
 
         Returns:
-            Data frame with two columns 'delay_ns' and 'value'
+            Data frame with columns 'delay_ns', 'num_pulses' and 'value'
+            The num_pulses column will only exist if multiple pulses
+            were requestd in the T2 experiment.
         """
-        xs = data['delay_ns']
+        delay = data['delay_ns']
         ones = data[1]
         zeros = data[0]
         pauli_expectation = (2 * (ones / (ones + zeros))) - 1.0
-        return pd.DataFrame({'delay_ns': xs, 'value': pauli_expectation})
+        if 'num_pulses' in data.columns:
+            return pd.DataFrame({
+                'delay_ns': delay,
+                'num_pulses': data['num_pulses'],
+                'value': pauli_expectation
+            })
+        return pd.DataFrame({'delay_ns': delay, 'value': pauli_expectation})
```

Comment:
missing type annotation on `data` argument up in the function signature (github won't let me comment there).

Reply:
Added.